agent/policyMgr/manager.go

```diff
@@ -91,6 +104,13 @@ func (a *policyManager) ManagePolicy(payload fleet.AgentPolicyRPCPayload) {
 		} else {
 			// attempt to apply the policy to the backend. status of policy application (running/failed) is maintained there.
 			be := backend.GetBackend(payload.Backend)
+			if updatePolicy {
+				// pktvisor has no update endpoint, so to update a policy it's necessary remove it and apply a new version
+				err := be.RemovePolicy(pd)
+				if err != nil {
+					a.logger.Warn("policy failed to remove", zap.String("policy_id", payload.ID), zap.String("policy_name", payload.Name), zap.Error(err))
+				}
+			}
 			a.applyPolicy(payload, be, &pd)
```

Comment:
what we should do instead is to include the `updatePolicy` bool as a parameter to `applyPolicy`. Then you can move the "remove first" logic to this pktvisor specific method implementation https://github.com/ns1labs/orb/blob/1a9ab8a563ba47c9a10f4546e259a4c44fe70e87/agent/backend/pktvisor/pktvisor.go#L149